Newspack/NewspackFramework/Extensions/FileManager+Helpers.swift

```diff
@@ -0,0 +1,34 @@
+import Foundation
+
+extension FileManager {
+
+    public func availableFileURL(for path: String, isDirectory: Bool, relativeTo relURL: URL?) -> URL {
+        var url = URL(fileURLWithPath: path, isDirectory: isDirectory, relativeTo: relURL).absoluteURL
+
+        if !fileExists(atPath: url.path) {
+            return url
+        }
+
+        var counter = 1
+        repeat {
+            let newPath = appendSuffix(suffix: String(counter), to: path)
+            url = URL(fileURLWithPath: newPath, isDirectory: isDirectory, relativeTo: relURL).absoluteURL
+            counter = counter + 1
+        } while fileExists(atPath: url.path)
+
+        return url
+    }
+
+    private func appendSuffix(suffix: String, to path: String) -> String {
+        guard path.contains(".") else {
+            return path + suffix
+        }
+
+        var parts = path.components(separatedBy: ".")
+        let str = parts[parts.count - 2] + suffix
```

Comment:
Is it possible that `parts` could ever not have a dot?

Reply:
Possibly!  But that scenario should be handled by the guard statement above.